### PR TITLE
[meta] package.json - Update Jest & jscodeshift

### PR DESCRIPTION
**Pull request by lb-**
* Resolves multiple npm audit issues - 11 vulnerabilities (9 moderate, 2 high)
* Manually ran tests locally while updating each version of `babel-jest` progressively and no issues found
* Ran the `npm run create` commands as documented after `jscodeshift` update and no issues found

After this update, as of today, there are now 0 vulnerabilities.

These are only development dependencies, nonetheless it's better to try to update these where possible as they still run on contributor machines.

#### Full audit report as per `main` branch at 4925ba8d0bf80a4b1d8e8645d310590bf1b40b64

```
➜ npm audit
# npm audit report

braces  <3.0.3
Severity: high
Uncontrolled resource consumption in braces - https://github.com/advisories/GHSA-grv7-fg5c-xmjg
fix available via `npm audit fix --force`
Will install jscodeshift@17.0.0, which is a breaking change
node_modules/@jest/transform/node_modules/braces
node_modules/jest-haste-map/node_modules/braces
node_modules/jest-message-util/node_modules/braces
node_modules/jscodeshift/node_modules/braces
node_modules/sane/node_modules/braces
  micromatch  <=4.0.7
  Depends on vulnerable versions of braces
  node_modules/@jest/transform/node_modules/micromatch
  node_modules/jest-haste-map/node_modules/micromatch
  node_modules/jest-message-util/node_modules/micromatch
  node_modules/jscodeshift/node_modules/micromatch
  node_modules/sane/node_modules/micromatch
    @jest/transform  <=24.9.0
    Depends on vulnerable versions of jest-haste-map
    Depends on vulnerable versions of jest-util
    Depends on vulnerable versions of micromatch
    node_modules/@jest/transform
      babel-jest  24.2.0-alpha.0 - 24.9.0
      Depends on vulnerable versions of @jest/transform
      node_modules/babel-jest
    anymatch  1.2.0 - 2.0.0
    Depends on vulnerable versions of micromatch
    node_modules/jest-haste-map/node_modules/anymatch
    node_modules/sane/node_modules/anymatch
      jest-haste-map  18.1.0 - 26.6.2
      Depends on vulnerable versions of anymatch
      Depends on vulnerable versions of jest-util
      Depends on vulnerable versions of micromatch
      Depends on vulnerable versions of sane
      node_modules/jest-haste-map
      sane  1.5.0 - 4.1.0
      Depends on vulnerable versions of anymatch
      Depends on vulnerable versions of micromatch
      node_modules/sane
    jest-message-util  18.5.0-alpha.7da3df39 - 24.9.0
    Depends on vulnerable versions of micromatch
    node_modules/jest-message-util
      @jest/fake-timers  <=24.9.0
      Depends on vulnerable versions of jest-message-util
      node_modules/@jest/fake-timers
        jest-util  24.2.0-alpha.0 - 24.9.0
        Depends on vulnerable versions of @jest/fake-timers
        node_modules/jest-util
    jscodeshift  0.3.20 - 0.13.1
    Depends on vulnerable versions of micromatch
    node_modules/jscodeshift


11 vulnerabilities (9 moderate, 2 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "@babel/plugin-transform-flow-strip-types": "^7.25.2",
     "@babel/register": "^7.24.6",
     "auto-changelog": "^2.4.0",
-    "babel-jest": "^24.9.0",
+    "babel-jest": "^29.7.0",
```
Comment (ljharb) [inline package.json:50]: actually we don’t use babel-jest anymore; it should be removed

**Review by ljharb — CHANGES_REQUESTED** [review]
Most audit warnings are false positives; having zero is nice but not a goal to optimize for.

That said, what are the breaking changes in jscodeshift? Tests don’t exercise it - it’s used to generate scaffolding for a new rule.

**Comment by lb-** [comment]
Thanks @ljharb 

I have confirmed I can run the two following scripts before and after the jscodeshift update without triggering any diffs in my local changes.

```
npm run create -- rule-name --author="Your name" --description="Description of rule"
npm run create -- no-marquee --author="Ethan Cohen <@evcohen>" --description="Enforce <marquee> elements are not used."
```

Example screenshot of files added

Screenshot 2024-10-19 at 8 52 44 AM

File: package.json (hunk 1 above)
Comment (lb-) [reply package.json]: ahh missed that, I wondered why the updates were so easy for Jest.

I have now removed that dependency and force pushed.

**Review by ljharb — APPROVED** [review]

**Comment by lb-** [comment]
Thanks @ljharb

**Pull request merged**